api/lib/block-listeners/base-node-subscription.js

```diff
@@ -169,6 +169,17 @@ class BaseNodeSubscription {
               tx
             )
           }
+
+          // TODO add claim events based on block events in Cosmos as claiming can happen also when undelegating/redelegating
```

Comment:
I didn't know this about Cosmos behavior. In which does undelegating or restaking trigger a claim rewards?

Reply:
in both txs you would claim all your rewards with that validator